Add Trino 395 release notes (#13975)
trinodb/trino: docs/src/main/sphinx/release/release-395.md

# Release 395 (7 Sep 2022)

## General

> * Significantly improve performance of inserts. ({issue}`13111`)

Reviewer: +1 that is too vague a release note
Also, it applies only to Hive, Delta and Iceberg.
Maybe "Improve performance of writes by automatically scaling writer count in tasks"
@sopel39 @gaurav8297 if this change has any implications for ppl who already manually tuned `task.writer-count` in their setup, please mention it here.

* Improve memory consumption of fault-tolerant execution. ({issue}`13855`)
* Improve memory consumption of aggregations. ({issue}`12512`)
* Improve performance of aggregations with decimals. ({issue}`13573`)
* Improve concurrency for large clusters. ({issue}`13934`, `13986`)
* Remove `information_schema.role_authorization_descriptors` table. ({issue}`11341`)
* Fix inaccurate results when using `SHOW CREATE TABLE` or `SHOW COLUMNS` on a
  table with columns that share a name with reserved keywords. ({issue}`13483`)

## ClickHouse connector

> * Fix query failure when renaming or dropping a column which should be quoted. ({issue}`13839`)

Author: I actually like the current note better - it's up front about what's being changed (a bug / query failure), and then it explains the circumstances in which the bug occurred. The connector wasn't quoting the column when it should've been quoted.

> * Fix query failure when renaming or dropping a column which should be quoted. ({issue}`13839`)

Reviewer 2: Yeah .. but its too vague ... " which should be quoted." .. which are those? I know that but its just not clear.

> * Fix query failure when renaming or dropping a column which should be quoted. ({issue}`13839`)

Reviewer 3: Is it "should be quoted" because the table name matches a SQL keyword, has non-legal characters, etc? If so, let's say that instead.

## Delta Lake connector

> * Add support for `ALTER TABLE ... RENAME TO` statement on Glue metastore. ({issue}`12985`)

Reviewer 2: either

```suggestion
* Add support for the `ALTER TABLE ... RENAME TO` statement with a Glue metastore. ({issue}`12985`)
```

or better

```suggestion
* Add support for `ALTER TABLE ... RENAME TO` statements  with a Glue metastore. ({issue}`12985`)
```

> * Enforce `delta.checkpoint.writeStatsAsJson` and

Reviewer 2: Specify what it does .. not just what properties are used..

  `delta.checkpoint.writeStatsAsStruct` table properties. ({issue}`12031`)

## Hive connector

* Improve performance of S3 Select when using CSV files as an input. ({issue}`13754`)

> * Ensure that the S3 KMS key is searched in the proper AWS region when S3

Reviewer 2: Does that fix a bug or is that a new feature? And what happened before?

  client-side encryption is used. ({issue}`13715`)

## Iceberg connector

* Fix creating metadata and manifest files with a URL-encoded name on S3 when
  the metadata location has trailing slashes. ({issue}`13759`)

## MariaDB connector

> * Fix query failure when renaming or dropping a column which should be quoted. ({issue}`13839`)

Reviewer 2: See other comment

## Memory connector

* Add support for table and column comments. ({issue}`13936`)

## MongoDB connector

* Fix query failure when filtering on columns with the `json` type. ({issue}`13536`)

## MySQL connector

* Fix query failure when renaming or dropping a column which should be quoted. ({issue}`13839`)

## Oracle connector

* Fix query failure when renaming or dropping a column which should be quoted. ({issue}`13839`)

## Phoenix connector

* Fix query failure when adding, renaming, or dropping a column which should be
  quoted. ({issue}`13839`)

## PostgreSQL connector

* Fix query failure when renaming or dropping a column which should be quoted. ({issue}`13839`)

## Prometheus connector

> * Add support for case-insensitive table name matching, which can be enabled

Reviewer 2: ```suggestion
* Add support for case-insensitive table name matching
```

  with the `prometheus.case-insensitive-name-matching` configuration property. ({issue}`8740`)

## Redshift connector

* Fix query failure when renaming or dropping a column which should be quoted. ({issue}`13839`)

## SingleStore (MemSQL) connector

* Fix query failure when renaming or dropping a column which should be quoted. ({issue}`13839`)

## SQL Server connector

* Fix query failure when renaming or dropping a column which should be quoted. ({issue}`13839`)

## SPI

* Add support for function resolution. ({issue}`12588`)

> * Remove `listAllRoleGrants` method from ConnectorMetadata. ({issue}`11341`)

Reviewer 2: ```suggestion
* Remove the `listAllRoleGrants` method from `ConnectorMetadata`. ({issue}`11341`)
```

# Release 395 (7 Sep 2022)

## General

> * Reduce memory consumption of fault-tolerant execution. ({issue}`13855`)

Reviewer 3: > ... when fault-tolerant execution is enabled.

* Reduce memory consumption of aggregations. ({issue}`12512`)
* Improve performance of aggregations with decimals. ({issue}`13573`)
* Improve concurrency for large clusters. ({issue}`13934`, `13986`)
* Remove `information_schema.role_authorization_descriptors` table. ({issue}`11341`)

> * Fix inaccurate results when using `SHOW CREATE TABLE` or `SHOW COLUMNS` on a

Reviewer 3: What does "inaccurate results" mean in this context?

  table with columns that use a reserved keyword as column name. ({issue}`13483`)

## ClickHouse connector

* Fix query failure when renaming or dropping a column which should be quoted. ({issue}`13839`)

## Delta Lake connector

* Add support for the `ALTER TABLE ... RENAME TO` statement with a Glue
  metastore. ({issue}`12985`)

> * Significantly improve performance of inserts by automatically scaling the

Reviewer: Please drop the "Significantly", that is a subjective term and we don't necessarily know how much things will improve, someone could be already using a high task.writer-count in their existing config.
Also, inserts -> writes as the improvement is not specific to INSERT query, CTAS query could benefit too.

Reviewer 2: Agreed

Reviewer: Please apply this comment to the hive and iceberg sections as well

  within a worker node. ({issue}`13111`)
* Enforce `delta.checkpoint.writeStatsAsJson` and
  `delta.checkpoint.writeStatsAsStruct` table properties to ensure table
  statistics are written in the correct format. ({issue}`12031`)

## Hive connector

> * Significantly improve performance of inserts by automatically scaling the

Reviewer 3: > Improve performance of inserts ...

  number of writers within a worker node. ({issue}`13111`)
* Improve performance of S3 Select when using CSV files as an input. ({issue}`13754`)
* Fix error where the S3 KMS key is not searched in the proper AWS region when
  S3 client-side encryption is used. ({issue}`13715`)

## Iceberg connector

> * Significantly improve performance of inserts by automatically scaling the

Reviewer 3: Same here

  number of writers within a worker node. ({issue}`13111`)
* Fix creating metadata and manifest files with a URL-encoded name on S3 when
  the metadata location has trailing slashes. ({issue}`13759`)

## MariaDB connector

* Fix query failure when renaming or dropping a column which should be quoted. ({issue}`13839`)

## Memory connector

* Add support for table and column comments. ({issue}`13936`)

## MongoDB connector

> * Fix query failure when filtering on columns with the `json` type. ({issue}`13536`)

Reviewer 3: > ... on columns of `json` type

or 

> ... on `json` columns

## MySQL connector

* Fix query failure when renaming or dropping a column which should be quoted. ({issue}`13839`)

## Oracle connector

* Fix query failure when renaming or dropping a column which should be quoted. ({issue}`13839`)

## Phoenix connector

* Fix query failure when adding, renaming, or dropping a column which should be
  quoted. ({issue}`13839`)

## PostgreSQL connector

* Fix query failure when renaming or dropping a column which should be quoted. ({issue}`13839`)

## Prometheus connector

* Add support for case-insensitive table name matching with the
  `prometheus.case-insensitive-name-matching` configuration property. ({issue}`8740`)

## Redshift connector

* Fix query failure when renaming or dropping a column which should be quoted. ({issue}`13839`)

## SingleStore (MemSQL) connector

* Fix query failure when renaming or dropping a column which should be quoted. ({issue}`13839`)

## SQL Server connector

* Fix query failure when renaming or dropping a column which should be quoted. ({issue}`13839`)

## SPI

> * Add support for function resolution. ({issue}`12588`)

Reviewer 3: > Add support for dynamic function resolution
